### PR TITLE
chore: Adds aria support for the switch

### LINKED ISSUE #1487: switch: Aria support
# Feature Request

Aria support for the switch webcomponent

Refrence the checkbox webcomponent

### PR DESCRIPTION
**Pull request by rowa-audil**
### <strong>Pull Request</strong>

Adds aria support for the switch

fixes #1487

#### Type of PR

Feature (non-breaking change which adds functionality)

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by lukasholzer — CHANGES_REQUESTED** [review]
please add a test if it works when the element is focused and you hit the space bar to toggle the aria-checked.

**Comment by rowa-audil** [comment]
@lukasholzer Added a test for SPACE and click

File: libs/fluid-elements/switch/src/lib/switch.spec.ts [hunk 1]
```diff
@@ -112,6 +112,42 @@ describe('Fluid switch', () => {
...
+
+    it('should have aria-checked set to true when the switch is checked', async () => {
+      fixture.checked = true;
+      await tick();
+      expect(fixture.hasAttribute('aria-checked')).toBeTruthy();
+      expect(fixture.getAttribute('aria-checked')).toBe('true');
+    });
+
+    it('should have aria-checked set to true when the switch is clicked', async () => {
+      const label = fixture.shadowRoot?.querySelector('label');
```
Comment (lukasholzer) [inline libs/fluid-elements/switch/src/lib/switch.spec.ts:136]: NIT: then you don't need the `label?.click()`

```suggestion
      const label = fixture.shadowRoot?.querySelector('label') as HTMLLabelElement;
```

**Review by bardosmisi — APPROVED** [review]
It looks good to me. 🦓

**Review by SkaillZ — APPROVED** [review]

**Review by SkaillZ — APPROVED** [review]

**Review by heartdisease — APPROVED** [review]

**Review by lukasholzer — APPROVED** [review]

File: libs/fluid-elements/switch/src/lib/switch.ts [hunk 2]
```diff
@@ -193,12 +193,51 @@ export class FluidSwitch extends LitElement {
...
+   */
+  @property({
+    type: String,
+    reflect: true,
+    attribute: 'aria-checked',
+  })
+  ariaChecked: string = 'false';
+
   /** First updated lifecycle */
   firstUpdated(props: Map<string | number | symbol, unknown>): void {
```
Comment (peter-affenzeller) [inline libs/fluid-elements/switch/src/lib/switch.ts:220]: you can use the `PropertyValues` from lit-element here for the type

File: libs/fluid-elements/switch/src/lib/switch.ts [hunk 3]
```diff
@@ -193,12 +193,51 @@ export class FluidSwitch extends LitElement {
...
     super.firstUpdated(props);
     this._inputElement = this.shadowRoot?.querySelector('input')!;
   }
 
+  /** Update lifecycle */
+  update(props: Map<string | number | symbol, unknown>): void {
```
Comment (peter-affenzeller) [inline libs/fluid-elements/switch/src/lib/switch.ts:226]: ... and here 🙂

File: libs/fluid-elements/switch/src/lib/switch.ts (hunk 2 above)
Comment (rowa-audil) [reply libs/fluid-elements/switch/src/lib/switch.ts:220]: Thank you :)

**Pull request merged**